tests/test_map.cpp

```diff
@@ -1,11 +1,22 @@
 #include <algorithm>
 #include <frozen/map.h>
+#include <functional>
 #include <iostream>
 #include <map>
 
 #include "bench.hpp"
 #include "catch.hpp"
 
+struct S {
+  int x;
+};
+constexpr inline bool operator==(S const &a, S const &b) { return a.x == b.x; }
+constexpr inline bool operator==(S const &a, int b) { return a.x == b; }
+constexpr inline bool operator==(int a, S const &b) { return a == b.x; }
+constexpr inline bool operator<(S const &a, S const &b) { return a.x < b.x; }
+constexpr inline bool operator<(S const &a, int b) { return a.x < b; }
+constexpr inline bool operator<(int a, S const &b) { return a < b.x; }
+
```

Comment:
can you move these riht before the concerned test case, for locality?

Reply:
`S` is not defined as transparent, is this intended? I would have made it transparent, and remove the overload that doesn't take an int parameter, to make sure the transparent version are used (?)

Reply:
`is_transparent` is an attribute to the `Compare` type, not the type itself. ie. `S` is not transparent, just a type that can be "transparently" compared to a different type, `int` in this case.

How do you make `S` "transparent"?